src/pushsource/_impl/helpers.py

```diff
@@ -85,3 +93,22 @@ def try_bool(value):
     elif value in ["", "0", "false", "no"]:
         return False
     raise ValueError("Expected a boolean, got %s" % repr(value))
+
+
+def force_https(url: str) -> str:
```

Comment:
You can't use inline type hints because the code has to be compatible with python2.  Currently, this is the cause of at least some of the CI failures.

Reply:
Oh wow that was careless of me to leave in there, good catch. I had hope that PyCharm's version compatibility checks would have warned me of this, but I'm becoming increasingly convinced they aren't very robust.